filters/builtin/dynamicbackendfilter.go

```diff
@@ -0,0 +1,157 @@
+// Instances expect one parameters: a header name.
+// Name: "setDynamicBackendHostFromHeader".
+//
+// If the header exists the value is put into the `StateBag`, additionally
+// `SetOutgoingHost()` is used to set the host header
+func NewSetDynamicBackendHostFromHeader() filters.Spec {
+	return &dynamicBackendFilter{typ: setDynamicBackendHostFromHeader}
+}
+
+// Returns a filter specification that is used to set dynamic backend scheme from a header.
+// Instances expect one parameters: a header name.
+// Name: "setDynamicBackendSchemeFromHeader".
+//
+// If the header exists the value is put into the `StateBag`
+func NewSetDynamicBackendSchemeFromHeader() filters.Spec {
+	return &dynamicBackendFilter{typ: setDynamicBackendSchemeFromHeader}
+}
+
+// Returns a filter specification that is used to set dynamic backend url from a header.
+// Instances expect one parameters: a header name.
+// Name: "setDynamicBackendUrlFromHeader".
+//
+// If the header exists the value is put into the `StateBag`, additionally
+// `SetOutgoingHost()` is used to set the host header if the header is a valid url
+func NewSetDynamicBackendUrlFromHeader() filters.Spec {
+	return &dynamicBackendFilter{typ: setDynamicBackendUrlFromHeader}
+}
+
+// Returns a filter specification that is used to set dynamic backend host.
+// Instances expect one parameters: a host name.
+// Name: "setDynamicBackendHost".
+//
+// The value is put into the `StateBag`, additionally
+// `SetOutgoingHost()` is used to set the host header
+func NewSetDynamicBackendHost() filters.Spec {
+	return &dynamicBackendFilter{typ: setDynamicBackendHost}
+}
+
+// Returns a filter specification that is used to set dynamic backend scheme.
+// Instances expect one parameters: a scheme name.
+// Name: "setDynamicBackendScheme".
+//
+// The value is put into the `StateBag`
+func NewSetDynamicBackendScheme() filters.Spec {
+	return &dynamicBackendFilter{typ: setDynamicBackendScheme}
+}
+
+// Returns a filter specification that is used to set dynamic backend url.
+// Instances expect one parameters: a url.
+// Name: "setDynamicBackendUrl".
+//
+// The value is put into the `StateBag`, additionally `SetOutgoingHost()`
+// is used to set the host header if the input provided is a valid url
+func NewSetDynamicBackendUrl() filters.Spec {
+	return &dynamicBackendFilter{typ: setDynamicBackendUrl}
+}
+
+func (spec *dynamicBackendFilter) Name() string {
+	switch spec.typ {
+	case setDynamicBackendHostFromHeader:
+		return SetDynamicBackendHostFromHeader
+	case setDynamicBackendSchemeFromHeader:
+		return SetDynamicBackendSchemeFromHeader
+	case setDynamicBackendUrlFromHeader:
+		return SetDynamicBackendUrlFromHeader
+	case setDynamicBackendHost:
+		return SetDynamicBackendHost
+	case setDynamicBackendScheme:
+		return SetDynamicBackendScheme
+	case setDynamicBackendUrl:
+		return SetDynamicBackendUrl
+	default:
+		panic("invalid type")
```

Comment:
panic is not good at runtime, better unknown type

Reply:
I think this one should be fine.

I didn't comment on this panic because the spec the values of typ can be set only from inside the package, but the spec initializers, so this panic basically cannot happen. Since something is required for the default, it is simply a matter of preference whether the impossible panic should happen here or delegated to somewhere else.

I quickly went through the other panics in the filters, most of them are similar to this case. Excaptions are the flowid and the compress packages, where the panics, if happen, happen only during the startup phase, meaning they prevent the program from starting up with an invalid state. So they should be fine, too.